test/src/test.ts

```diff
@@ -0,0 +1,186 @@
+import nunjucks, { render } from 'nunjucks';
+import util from 'util';
+import path from 'path';
+import * as fs from 'fs';
+
+describe('catalog model template tests', () => {
+    // Test default values with minimum set of templated files (catalog-info.yaml and index.md)
+    test('001-model-required-values-only', () => {
+        let modelTemplateValues = {
+            "values": {
+                "modelName": "ibm-granite-8b",
+                "modelDescription": "The IBM Granite 8b code instruct model, deployed on vLLM",
+                "modelURL": "https://model-server.example.com",
+                "modelServerName": "vllm",
+                "modelServerAPIURL": "https://api.model-server.example.com",
+                "modelResourceURL": "https://huggingface.co/ibm-granite/granite-3.1-8b-instruct",
+                "modelLicenseURL": "https://huggingface.co/datasets/choosealicense/licenses/blob/main/markdown/apache-2.0.md",
+                "modelLicense": "Apache-2",
+                "modelTags": ["granite", "ibm", "huggingface", "llm"],
+                "modelOwner": "rhdh-user",
+                "modelSupport": "https://github.com/ibm-granite/granite-3.1-language-models",
+                "modelDeveloper": "IBM Research"
```

Comment:
quick question, how does nunjucks resolve `modelDeveloper` (for example) when its rendering the file

Reply:
It renders as "IBM Research"

<img width="134" alt="Screenshot 2025-01-28 at 11 12 20 AM" src="https://github.com/user-attachments/assets/d86000e2-6974-48ba-829a-bd20509961c7" />

Reply:
In terms of how it resolves the values, I think it just checks for matching values in the `values` json object we pass in to it. E.g. if it see `values.modelDeveloper`, it'll check for a corresponding value in the json object